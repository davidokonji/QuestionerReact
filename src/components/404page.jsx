import React, { Fragment } from 'react';
import Header from './Landingheader';

const NotFound = (props) => {
  const { text } = props;
  return (
    <Fragment>
      <Header
        page="landingpage"
        navClass="nav_header"
      />
      <div id="notfound">
        <div className="notfound">
          <div className="notfound-404">
            <h3>
            Oops!
              {
              text || ' Page '
            }
              not found

            </h3>
            <h1>
              <span>4</span>
              <span>0</span>
              <span>4</span>
            </h1>
          </div>
          <h2>
          we are sorry, page you requested was not found
          </h2>
          <a href="/" className="homeLink">home</a>
        </div>
      </div>
    </Fragment>
  );
};

export default NotFound;
