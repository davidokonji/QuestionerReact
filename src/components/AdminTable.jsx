import React from 'react';
import MeetupList from './MeetupList';

const AdminTable = ({ data, toggle }) => (

  // const deleteMeeteps = ( id) => {
  //     this.props.delete(id)
  // }
  <div className='col-11 mx-4 mx-md-0 mt-4 mt-md-0 col-md-9 shadow p-0 table-responsive-sm'>
    <table className='table table-hover p-0'>
      <thead className='row mx-0'>
        <div className='col-10 bg-light p-3'>
            Available Meetup(s)
        </div>
        <div className='col-2 bg-light p-3'>
            Operation
        </div>
      </thead>
      <tbody className='row p-2 col-12 ml-0 '>
        {
          data.map(meetup => (
            <MeetupList
              id={meetup.id}
              title={meetup.title}
              location={meetup.location}
              tags={meetup.tags}
              happeningOn={meetup.happeningOn}
              images={meetup.images}
              key={meetup.id}
              // delete={delteMeetups}
            //   deleteDialog={toggle}
            />
          ))
      }
      </tbody>
    </table>
  </div>
);

export default AdminTable;
