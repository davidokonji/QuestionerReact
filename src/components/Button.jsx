import React from 'react';

const Button = (props) => {
  const { text, event, styles } = props;
  return (
    <button type="button" onClick={event} className={styles}>
      {text}
    </button>
  );
};

export default Button;
