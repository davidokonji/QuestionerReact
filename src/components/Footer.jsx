import React from 'react';
import '../style/footer.scss';

const Footer = () => {
  const date = new Date().getFullYear();
  return (
    <React.Fragment>
      <footer>
      &copy;
        {date}

      </footer>
    </React.Fragment>
  );
};

export default Footer;
