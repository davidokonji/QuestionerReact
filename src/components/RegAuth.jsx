import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { toast } from 'react-toastify';
import { RegisterAction, clear } from '../actions/UserAction';

class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      email: '',
      username: '',
      password: '',
      othername: '',
      phonenumber: ''
    };
  }

  onPasswordChange = (e) => {
    const password = e.target.value;
    if (password && password.match(/^(?=.*[a-zA-Z])(?=.*[0-9])/)) {
      this.setState({ password });
    } else {
      this.setState({
        password,
      });
    }
  };

  handleSignup = (e) => {
    e.preventDefault();

    const {
      firstname,
      lastname,
      username,
      email,
      password,
      phonenumber,
      othername
    } = this.state;
    const { userRegister } = this.props;

    const data = {
      firstname,
      lastname,
      username,
      email,
      password,
      phonenumber,
      othername
    };
    userRegister(data);
  }

  handleInputChange = (event) => {
    const { value, name } = event.target;
    this.setState({
      [name]: value,
    });
  }

  resetForm = () => {
    this.setState({
      firstname: '',
      lastname: '',
      email: '',
      username: '',
      password: '',
      othername: '',
      phonenumber: ''
    });
  }

  render() {
    const {
      firstname,
      lastname,
      email,
      username,
      password,
      othername,
      phonenumber
    } = this.state;

    const {
      user: {
        message = ''
      },
      redirect = false,
      history: { push },
      clearError
    } = this.props;
    if (redirect) {
      toast(message, {
        position: toast.POSITION.TOP_CENTER,
        type: toast.TYPE.SUCCESS,
        onClose: () => push('/dashboard')
      });
    } else if (redirect === false && message.length !== 0) {
      toast(message, {
        position: toast.POSITION.TOP_CENTER,
        type: toast.TYPE.WARNING,
        className: 'rotateY animated'
      });
      this.resetForm();
      clearError();
    }

    return (
      <form method='post' onSubmit={this.handleSignup}>
        <div className='signup-form' id='signup-form'>
          <div className='double'>
            <input
              type='text'
              placeholder='First Name'
              name='firstname'
              value={firstname}
              onChange={this.handleInputChange}
              className='input form_sm'
              required
            />
            <input
              type='text'
              placeholder='Last Name'
              name='lastname'
              value={lastname}
              onChange={this.handleInputChange}
              className='input form_sm'
              required
            />
          </div>
          <div className='double'>
            <input
              type='text'
              placeholder='Username'
              value={username}
              onChange={this.handleInputChange}
              name='username'
              className='input form_sm'
              required
            />
          </div>
          <div className='double'>
            <input
              type='text'
              placeholder='other name'
              value={othername}
              onChange={this.handleInputChange}
              name='othername'
              className='input form_sm'
              required
            />
          </div>
          <div className='double'>
            <input
              type='text'
              placeholder='Phone number'
              value={phonenumber}
              onChange={this.handleInputChange}
              name='phonenumber'
              className='input form_sm'
              required
            />
          </div>
          <input
            type='email'
            placeholder='Email Address'
            name='email'
            value={email}
            onChange={this.handleInputChange}
            className='input form_lg'
            required
          />
          <input
            type='password'
            placeholder='Password'
            name='password'
            value={password}
            onChange={this.onPasswordChange}
            className='input form_lg'
            required
          />
          <div>
            <button className='btn' type='submit'>
                    Create account
            </button>
          </div>
        </div>
      </form>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.auth,
  redirect: state.auth.redirect
});

const mapDispatchToProps = {
  userRegister: RegisterAction,
  clearError: clear
};

const RegisterConnected = withRouter(connect(mapStateToProps, mapDispatchToProps)(Register));
export {
  Register,
  RegisterConnected
};
