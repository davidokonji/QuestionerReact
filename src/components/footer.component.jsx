import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Footer = () => (
  <footer className="footer">
    <div className="footerBottom">
      <p>
       Created with
        {<FontAwesomeIcon icon="heart" size="sm" color="red" style={{ paddingLeft: '1rem', paddingRight: '1rem' }} />}
       by Okonji Chukwunonso
      </p>
    </div>
  </footer>
);

export default Footer;
