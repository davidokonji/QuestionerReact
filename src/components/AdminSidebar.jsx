import React, { Fragment, Component } from 'react';
import { connect } from 'react-redux';
import { WithContext as ReactTags } from 'react-tag-input';
import {
  Form,
  FormGroup,
  Label,
  Input,
  FormText,
} from 'reactstrap';
import Button from './Button';
import { addMeetup } from '../actions';
import Modal from './common/modal';

const KeyCodes = {
  comma: 188,
  enter: 13,
};

const delimiters = [KeyCodes.comma, KeyCodes.enter];
export class AdminSidebar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tagList: [
      ],
      suggestions: [
        { id: 'technology', text: 'technology' },
        { id: 'social', text: 'social' },
      ],
      topic: '',
      location: '',
      happeningOn: '',
      images: '',
      imagePreview: ''
    };
  }

  handleOnChange = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  }

  imageHandler = ({ target: { files } }) => {
    this.setState({
      images: files[0],
      imagePreview: URL.createObjectURL(files[0]),
    });
  }


  handleSubmit = (e) => {
    e.preventDefault();
    const { createMeetup } = this.props;
    const {
      topic,
      location,
      happeningOn,
      tagList,
      images
    } = this.state;
    const filteredTags = tagList.map(tag => tag.text);
    const tags = filteredTags.join();
    const data = {
      topic,
      location,
      happeningOn,
      tags,
      images
    };

    createMeetup(data);
  }

  handleAddition = (tag) => {
    this.setState(state => ({ tagList: [...state.tagList, tag] }));
  }

  handleDrag = (tag, currPos, newPos) => {
    const { tagList } = this.state;
    const setTags = new Set([...tagList]);
    const newTags = setTags.slice();

    newTags.splice(currPos, 1);
    newTags.splice(newPos, 0, tag);

    this.setState({ tagList: newTags });
  }

  handleDelete = (i) => {
    const { tagList } = this.state;
    this.setState({
      tagList: tagList.filter((tag, index) => index !== i),
    });
  }

  render() {
    const { toggle, open } = this.props;
    const {
      tagList,
      suggestions,
      imagePreview,
      topic
    } = this.state;
    const placeholder = 'Add meetup Tags';
    return (
      <Fragment>
        <div className='col-12 col-md-3'>
          <div className='p-5 shadow-sm'>
            <Button styles='button p-3' text='Create Meetup' event={toggle} />
          </div>
        </div>
        <Modal toggle={toggle} open={open} className='mt-5'>
          <Form method='post' onSubmit={this.handleSubmit} encType='multipart/form-data'>
            {
              imagePreview && (
                <div>
                  <img src={imagePreview} alt={topic} style={{ height: '30rem' }} />
                </div>
              )
            }
            <FormGroup>
              <Input
                type='file'
                name='image'
                max={1}
                accept='image/*'
                onChange={this.imageHandler}
              />
              <FormText color='muted'>
              Add meetup images
              </FormText>
            </FormGroup>
            <FormGroup>
              <Input
                type='text'
                name='topic'
                placeholder='Add Topic'
                onChange={this.handleOnChange}
                className='px-3'
              />
            </FormGroup>
            <FormGroup>
              <Input
                type='text'
                name='location'
                placeholder='Add Location'
                onChange={this.handleOnChange}
              />
            </FormGroup>
            <FormGroup>
              <Input
                type='date'
                name='happeningOn'
                placeholder='Add Date'
                onChange={this.handleOnChange}
              />
            </FormGroup>
            <FormGroup>
              <div id='app'>
                <ReactTags
                  tags={tagList}
                  suggestions={suggestions}
                  handleDelete={this.handleDelete}
                  handleAddition={this.handleAddition}
                  handleDrag={this.handleDrag}
                  delimiters={delimiters}
                  placeholder={placeholder}
                  inline={false}
                />
              </div>
            </FormGroup>
            <button type='submit' className='btn' onClick={toggle}>Create Meetup</button>
          </Form>
        </Modal>
      </Fragment>
    );
  }
}
const mapDispatchToProps = {
  createMeetup: addMeetup
};


export default connect(null, mapDispatchToProps)(AdminSidebar);
