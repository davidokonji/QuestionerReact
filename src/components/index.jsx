import React from 'react';
import Navbar from './Navbar';
import Slider from './slider';
import Footer from './Footer';

const App = () => {
  return (
    <React.Fragment>
      <Navbar />
      <Slider />
      <Footer />
    </React.Fragment>
  );
};

export default App;
