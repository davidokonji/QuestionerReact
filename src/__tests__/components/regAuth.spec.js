import React from 'react';
import { shallow } from 'enzyme';
import { Register } from '../../components/RegAuth';

let wrapper;
const user = {
  message: ''
};
const history = {
  push: jest.fn()
};

describe('Register Compnent', () => {
  beforeAll(() => {
    const userRegister = jest.fn();

    wrapper = shallow(<Register user={user} userRegister={userRegister} history={history} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle firstname change', () => {
    const value = 'john';

    wrapper.find('input').at(0).simulate('change', {
      target: { value }
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle firstname change', () => {
    const value = 'snow';

    wrapper.find('input').at(1).simulate('change', {
      target: { value }
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle password change', () => {
    const value = 'password1234';

    wrapper.find('input').at(4).simulate('change', {
      target: { value }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle password change', () => {
    const value = 'password';

    wrapper.find('input').at(4).simulate('change', {
      target: { value }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle form submit', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper).toMatchSnapshot();
  });
});
