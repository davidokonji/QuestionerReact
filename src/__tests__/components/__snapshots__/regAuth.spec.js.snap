// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    className="signup-form"
    id="signup-form"
  >
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_sm"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="othername"
        onChange={[Function]}
        placeholder="other name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="phonenumber"
        onChange={[Function]}
        placeholder="Phone number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <input
      className="input form_lg"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
    <input
      className="input form_lg"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Create account
      </button>
    </div>
  </div>
</form>
`;

exports[`Register Compnent should handle firstname change 1`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    className="signup-form"
    id="signup-form"
  >
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_sm"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="othername"
        onChange={[Function]}
        placeholder="other name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="phonenumber"
        onChange={[Function]}
        placeholder="Phone number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <input
      className="input form_lg"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
    <input
      className="input form_lg"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Create account
      </button>
    </div>
  </div>
</form>
`;

exports[`Register Compnent should handle firstname change 2`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    className="signup-form"
    id="signup-form"
  >
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_sm"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="othername"
        onChange={[Function]}
        placeholder="other name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="phonenumber"
        onChange={[Function]}
        placeholder="Phone number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <input
      className="input form_lg"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
    <input
      className="input form_lg"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Create account
      </button>
    </div>
  </div>
</form>
`;

exports[`Register Compnent should handle form submit 1`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    className="signup-form"
    id="signup-form"
  >
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_sm"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="othername"
        onChange={[Function]}
        placeholder="other name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="phonenumber"
        onChange={[Function]}
        placeholder="Phone number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <input
      className="input form_lg"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
    <input
      className="input form_lg"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Create account
      </button>
    </div>
  </div>
</form>
`;

exports[`Register Compnent should handle password change 1`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    className="signup-form"
    id="signup-form"
  >
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_sm"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="othername"
        onChange={[Function]}
        placeholder="other name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="phonenumber"
        onChange={[Function]}
        placeholder="Phone number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <input
      className="input form_lg"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
    <input
      className="input form_lg"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Create account
      </button>
    </div>
  </div>
</form>
`;

exports[`Register Compnent should handle password change 2`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    className="signup-form"
    id="signup-form"
  >
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="input form_sm"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="othername"
        onChange={[Function]}
        placeholder="other name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="double"
    >
      <input
        className="input form_sm"
        name="phonenumber"
        onChange={[Function]}
        placeholder="Phone number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <input
      className="input form_lg"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
    <input
      className="input form_lg"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Create account
      </button>
    </div>
  </div>
</form>
`;
